src/skmultiflow/meta/learn_nse.py

```diff
@@ -0,0 +1,286 @@
+
+                else:
+                    # First run! train the classifier on the instances with the same weight
+                    self.instance_weights = np.ones(mt) / mt
+
+                    # classifier.fit(self.X_batch, self.y_batch, sample_weight=self.instanceWeights)
+                    self.__fit(classifier, self.X_batch, self.y_batch, sample_weight=self.instance_weights,
+                               classes=self.classes)
+
+                self.ensemble.append(classifier)
+                self.bkts.append([])
+                self.wkts.append([])
+                self.ensemble_weights = []
+
+                t = len(self.ensemble)
+                max_error = None
+                error_index = None
+
+                # Evaluate all existing classifiers on the new dataset
+                for k in range(1, t + 1):
+                    pred = self.ensemble[k - 1].predict(self.X_batch)
+                    ekt = np.sum(self.instance_weights[pred != self.y_batch])
+
+                    if k == t and ekt > 0.5:
+                        # Generate a new classifier
+                        classifier = cp.deepcopy(self.base_estimator)
+                        self.__fit(classifier, self.X_batch, self.y_batch, sample_weight=self.instance_weights,
+                                   classes=self.classes)
+                        self.ensemble[k - 1] = classifier
+                    elif ekt > 0.5:
+                        ekt = 0.5
+
+                    # Storing the index of the classifier with higher error in case of error-based pruning
+                    if max_error is None or ekt > max_error:
+                        max_error = ekt
+                        error_index = k
+
+                    # Normalize errors
+                    bkt = ekt / (1 - ekt)
+
+                    # store normalized error for this classifier
+                    nbkts = self.bkts[k - 1]
+                    nbkts.append(bkt)
+
+                    # compute the weighted normalized errors for kth classifier h_k
+                    wkt = 1.0 / (1.0 + np.exp(-self.slope * (t - k - self.crossing_point)))
+                    weights = self.wkts[k - 1]
+                    weights.append(wkt / (np.sum(weights) + wkt))
+
+                    sbkt = np.sum(np.array(nbkts) * np.array(weights)) + 1e-50
+
+                    # Calculate classifier voting weights
+                    self.ensemble_weights.append(np.log(1.0 / sbkt))
+
+                # Ensemble pruning
+
+                if self.pruning == 'age' and t > self.ensemble_size:
+                    # Age-based
+                    self.ensemble.pop(0)
+                    self.ensemble_weights.pop(0)
+                    self.bkts.pop(0)
+                    self.wkts.pop(0)
+                elif self.pruning == 'error' and t > self.ensemble_size:
+                    # Error-based
+                    self.ensemble.pop(error_index - 1)
```

Comment:
This can result in an exception if `error_index` has not been set.

Reply:
the error_index will always be set after we run the code
 ```
if ekt > max_error:
    max_error = ekt
    error_index = k
```